### PR TITLE
update versions bbased on git tags

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
We currently run the update script as part of the release process. This causes an issue (highlighted by @cocreature in #6560): the update script currently relies on querying the docs website to get the list of versions, but the docs website is updated on a cron, which means by the time we run the update script it is not up-to-date yet.

This PR changes the update script to instead rely on the local git repo and its list of tags, as that is updated synchronously.

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: compatibility/update-versions.sh [hunk 1]
```diff
@@ -6,7 +6,10 @@
 cd "$(dirname "$0")"
 eval "$(../dev-env/bin/dade-assist)"
 
-bazel run //versions:update-versions -- -o $PWD/versions.bzl
+# Note: not using Bazel run so we actually have access to local files to run
+# `git tag`
+bazel build //versions:update-versions
+bazel-bin/versions/update-versions -o $PWD/versions.bzl
```
Comment (cocreature) [inline compatibility/update-versions.sh:12]: Perfectly happy to change it so just an fyi: `bazel run --run_under "cd $PWD &&" …` keeps the working directory.

File: compatibility/versions/UpdateVersions.hs [hunk 2]
```diff
@@ -168,15 +162,16 @@ optsParser :: Parser Opts
 optsParser = Opts
   <$> strOption (short 'o' <> help "Path to output file")
 
+getVersionsFromTags :: IO (Set Version, Set Version)
+getVersionsFromTags = do
+    tags <- lines <$> System.Process.readProcess "git" ["tag"] ""
+    let versions = Set.fromList $ rights $ map (SemVer.fromText . T.pack . tail) tags
```
Comment (cocreature) [inline compatibility/versions/UpdateVersions.hs:168]:
```suggestion
    let versions = Set.fromList $ rights $ mapMaybe (fmap (SemVer.fromText . T.pack) . stripPrefix "p") tags
```
A bit more explicit. I’m also fine with just a comment as to why we call `tail`. Note that `tail` crashes on an empty list (not that this matters here).

Comment (garyverhaegen-da) [reply compatibility/versions/UpdateVersions.hs:168]: I imagine you meant `stripPrefix "v"`?

Comment (cocreature) [reply compatibility/versions/UpdateVersions.hs:168]: Eh yeah, sorry about that.

Comment (garyverhaegen-da) [reply compatibility/versions/UpdateVersions.hs]: Made the change.

File: compatibility/update-versions.sh (hunk 1 above)
Comment (garyverhaegen-da) [reply compatibility/update-versions.sh]: Done.

**Pull request merged**